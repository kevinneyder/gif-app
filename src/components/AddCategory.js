import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = ( e ) => {
        setinputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 0){
            setCategories(oldArray => [inputValue, ...oldArray]);
            setinputValue('');
        }
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <input
                type  = 'text'
                value = { inputValue }
                onChange = { ( handleInputChange ) }
            />   
        </form>
    )
}

AddCategory.propType = {
    setCategories: PropTypes.func.isRequired
}