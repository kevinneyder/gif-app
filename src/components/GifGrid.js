import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data: gif, loadding} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn" >{category}</h3>
            { loadding && <p className="animate__animated animate__flash">Loading...</p> }
            <div className="card-grid">
                {
                    gif.map((gif) => (
                        <GifGridItem
                            key = {gif.id}
                            {...gif}
                        />
                    ))
                }
            </div>
        </>
    )
}
