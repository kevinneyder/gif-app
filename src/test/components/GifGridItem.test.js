import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Prueba Gif grid', () => {
    
    test('Debe mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();
    })
    

})
