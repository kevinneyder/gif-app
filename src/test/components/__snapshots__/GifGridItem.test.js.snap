// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Gif grid Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
